File: typescript/src/points/polarpoint.ts
interface PolarPoint {
    len: number,
    angle: number,
    computeX: () => number
    computeY: () => number
}

function newPolarPoint (len: number, angle: number): PolarPoint {
    return {
        len,
        angle,

        computeX: function (): number {
            return Math.round(len * Math.cos(angle))
        },

        computeY: function (): number {
            return Math.round(len * Math.sin(angle))
        }
    }
}

export { PolarPoint, newPolarPoint }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/points/PolarPoint.java
package edu.cmu.cs.cs214.lab02.points;

public class PolarPoint implements Point {
    public double len;
    public double angle;

    public PolarPoint (double len, double angle) {
        this.len = len;
        this.angle = angle;
    }

    public int computeX() {
        return (int) (this.len * Math.cos(this.angle));
    }

    public int computeY() {
        return (int) (this.len * Math.sin(this.angle));
    }
}
